typings/objection/index.d.ts

```diff
@@ -198,6 +213,7 @@ declare namespace Objection {
     to: string | Reference | (string | Reference)[];
     modelClass?: ModelClass<any> | string;
     extra?: string[] | object;
+    beforeInsert?: (model: Model, context: QueryContext) => void;
   }
```

Comment:
This method can be async, so maybe the return value should be `void | Promise<any>` or something? I'm not sure how other async hooks have been typed.